streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -217,4 +228,24 @@ public void setUncaughtExceptionHandler(Thread.UncaughtExceptionHandler eh) {
             thread.setUncaughtExceptionHandler(eh);
     }
 
+    /**
+     * Get a facade wrapping the {@link org.apache.kafka.streams.processor.StateStore} instances
+     * with the provided storeName and accepted by {@link QueryableStoreType#accepts(StateStore)}.
+     * The returned object can be used to query the {@link org.apache.kafka.streams.processor.StateStore} instances
+     * @param storeName             name of the store to find
+     * @param queryableStoreType    accept only stores that are accepted by {@link QueryableStoreType#accepts(StateStore)}
+     * @param <T>                   return type
+     * @return  A facade wrapping the {@link org.apache.kafka.streams.processor.StateStore} instances
+     */
+    public <T> T store(final String storeName, final QueryableStoreType<T> queryableStoreType) {
+        validateIsRunning();
+        return queryableStoreProvider.getStore(storeName, queryableStoreType);
+    }
+
+    private void validateIsRunning() {
```

Comment:
Why is this check required? After given the state handle to the user, she might close the application and use the store... (ie, same problem).
Why extra method for check? Only used once.